public/API/API_GetGameProgression.php

```diff
@@ -0,0 +1,177 @@
+ *   string?    Type                             type of the achievement (progression/win_condition/missable/null)
+ *   string     BadgeName                        unique identifier of the badge image for the achievement
+ *   int        NumAwarded                       number of times the achievement has been awarded
+ *   int        NumAwardedHardcore               number of times the achievement has been awarded in hardcore
+ *   int        TimesUsedInUnlockMedian          number of unlocks analyzed for the MedianTimeToUnlock metric
+ *   int        TimesUsedInHardcoreUnlockMedian  number of unlocks analyzed for the MedianTimeToUnlockHardcore metric
+ *   int        MedianTimeToUnlock               median number of seconds required to unlock this achievement from starting to play the game
+ *   int        MedianTimeToUnlockHardcore       median number of seconds required to unlock this achievement in hardcore from starting to play the game
+ */
+
+use App\Models\Achievement;
+use App\Models\Game;
+use App\Models\PlayerAchievement;
+use App\Models\PlayerGame;
+use App\Models\PlayerProgressReset;
+use App\Models\PlayerSession;
+use App\Platform\Enums\PlayerProgressResetType;
+
+$gameId = (int) request()->query('i');
+$preferHardcore = (int) request()->query('h');
+
+$game = Game::with('system')->find($gameId);
+if (!$game) {
+    return response()->json([], 404);
+}
+
+// ===== basic game information =====
+$coreSet = $game->gameAchievementSets()->core()->first()?->achievementSet;
+
+$response = [
+    'ID' => $game->ID,
+    'Title' => $game->Title,
+    'ConsoleID' => $game->system->ID,
+    'ConsoleName' => $game->system->Name,
+    'ImageIcon' => $game->ImageIcon,
+    'NumDistinctPlayers' => $game->players_total,
+    'TimesUsedInBeatMedian' => $game->times_beaten,
+    'TimesUsedInHardcoreBeatMedian' => $game->times_beaten_hardcore,
+    'MedianTimeToBeat' => $game->median_time_to_beat,
+    'MedianTimeToBeatHardcore' => $game->median_time_to_beat_hardcore,
+    'TimesUsedInCompletionMedian' => $coreSet?->times_completed ?? 0,
+    'TimesUsedInMasteryMedian' => $coreSet?->times_completed_hardcore ?? 0,
+    'MedianTimeToComplete' => $coreSet?->median_time_to_complete,
+    'MedianTimeToMaster' => $coreSet?->median_time_to_complete_hardcore,
+    'NumAchievements' => $game->achievements_published,
+    'Achievements' => [],
+];
+
+$achievements = $game->achievements()->published()->get();
+$achievementIds = $achievements->pluck('ID');
+$unlock_times = [];
+$unlock_hardcore_times = [];
+foreach ($achievementIds as $achievementId) {
+    $unlock_times[$achievementId] = [];
+    $unlock_hardcore_times[$achievementId] = [];
+}
+
+// ===== process the 100 most recent players to earn at least half of the achievements in the set =====
+$recentPlayerIds = PlayerGame::query()
+    ->where('game_id', $game->ID)
+    ->where($preferHardcore ? 'achievements_unlocked_hardcore' : 'achievements_unlocked', '>=', $game->achievements_published / 2)
```

Comment:
Intellectual curiosity: why limit this to only players who have earned at least half the achievements? Could this skew statistics for difficult games where most players earn <50%?